src/gtc/daceir.py

```diff
@@ -827,20 +834,27 @@ class Temporary(FieldDecl):
     pass
 
 
-class ComputationNode(LocNode):
-    # mapping connector names to tuple of field name and access info
-    read_memlets: List[Memlet]
-    write_memlets: List[Memlet]
-
-    @validator("read_memlets", "write_memlets")
-    def unique_connectors(cls, node: List[Memlet]):
-        conns: Dict[SymbolRef, Set[SymbolRef]] = {}
+def _unique_connectors(*, field: str) -> datamodels.FieldValidator:
+    def _validator(
+        self: "ComputationNode", attribute: datamodels.Attribute, node: List[Memlet]
+    ) -> None:
+        conns: Dict[eve.SymbolRef, Set[eve.SymbolRef]] = {}
         for memlet in node:
             conns.setdefault(memlet.field, set())
             if memlet.connector in conns[memlet.field]:
                 raise ValueError(f"Found multiple Memlets for connector '{memlet.connector}'")
             conns[memlet.field].add(memlet.connector)
-        return node
+
+    return datamodels.validator(field)(_validator)
+
+
+class ComputationNode(LocNode):
+    # mapping connector names to tuple of field name and access info
+    read_memlets: List[Memlet]
+    write_memlets: List[Memlet]
+
+    unique_write_connectors = _unique_connectors(field="write_memlets")
+    unique_read_connectors = _unique_connectors(field="write_memlets")
```

Comment:
I think there is a copy-paste error here because `field="write_memlets"` is duplicated.

Besides that, I think there is no need for the `_unique_connectors` utility function to reuse the same validator for different fields. IIRC, nesting the `datamodels.validator` decorator should work just fine.

```suggestion
    @datamodels.validator("read_memlets")
    @datamodels.validator("write_memlets")
    def _validator(...):
       ....
```

Reply:
Thanks!